From a502123a847decefb3a283784dc2edb4e42d932f Mon Sep 17 00:00:00 2001
From: Nate Graham <nate@kde.org>
Date: Mon, 13 Feb 2023 17:40:52 -0700
Subject: [PATCH] Sort connector names for multi-screen size/position keys

A string including the connector names is used in the config file key for
storing per-screen-arrangement size and position data, which means we
need this string to be consistent for the same screen arrangement. But
connector order is non-deterministic. We need to sort the list to keep a
consistent order and avoid losing multi-screen size and position data.

CCBUG: 460260
(cherry picked from commit 3f84ef821d76cd7c646d4328b4627eb70dae4ab3)
---
 src/gui/kwindowconfig.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/gui/kwindowconfig.cpp b/src/gui/kwindowconfig.cpp
index 38721d5d..5e1de4ca 100644
--- a/src/gui/kwindowconfig.cpp
+++ b/src/gui/kwindowconfig.cpp
@@ -33,6 +33,12 @@ static QString allConnectedScreens()
     for (auto screen : screens) {
         names << screen->SCREENNAME();
     }
+    // A string including the connector names is used in the config file key for
+    // storing per-screen-arrangement size and position data, which means we
+    // need this string to be consistent for the same screen arrangement. But
+    // connector order is non-deterministic. We need to sort the list to keep a
+    // consistent order and avoid losing multi-screen size and position data.
+    names.sort();
     return names.join(QLatin1Char(' '));
 }
 
-- 
GitLab

